src/Message.ts

```diff
@@ -279,6 +279,13 @@ export class DecodedMessage {
     this.content = content
     this.contentTopic = contentTopic
   }
+}
+
+export class DecodedMessage extends DecodedMessageBase {
```

Comment:
Curious, why you decided to split off the base class?

Reply:
Ya. I would just put this into the `DecodedMessage` type. I don't see what extending the types is buying you here

Reply:
Wasn't my first choice. Perhaps theres an easier way...

The `DecodedMessage` constructor takes a `DecodedMessage` as a parameter (rather than a defined interface. :: https://github.com/xmtp/xmtp-js/blob/main/src/Message.ts#L259-L270

When adding a function to `DecodedMessage` the type is changed and all previous instantiations fail with:
`Argument of type '{ id: string; messageVersion: "v1"; senderAddress: string; recipientAddress: string | undefined; sent: Date; content: any; contentType: ContentTypeId; contentTopic: string; conversation: Conversation; error: Error | undefined; }' is not assignable to parameter of type 'DecodedMessage'.
  Property 'isOutboundMessage' is missing in type '{ id: string; messageVersion: "v1"; senderAddress: string; recipientAddress: string | undefined; sent: Date; content: any; contentType: ContentTypeId; contentTopic: string; conversation: Conversation; error: Error | undefined; }' but required in type 'DecodedMessage'.ts(2345)`
  
- Creating a `DecodedMessageParams` interface requires both the Interface and Class requires properties to be defined in both and kept in sync.
- There looked to be a work around with  class/interface merging but implementing it was messier.   
- Flattening the constructor would be a breaking change.

Anything I am missing?

Reply:
Think you can use some fancy typing to get around that problem. If you used something like `Omit<DecodedMessage, 'isOutboundMessage'>` in the constructor would work.

Reply:
![image](https://user-images.githubusercontent.com/473256/205745624-9cc7f5f5-de82-407e-8900-259825056d45.png)